### PR TITLE
Run iOS gen_snapshot as x86_64 arch

### PR DESCRIPTION
**Pull request by cbracken**
We are about to begin building gen_snapshot as a multi-arch binary,
which when run as x86_64 will generate arm64 AOT output, and when run as
i386 will generate armv7 AOT output.

Currently, gen_snapshot is an x86_64 binary, so this change is
effectively preventative in nature, and is a no-op with the current
snapshotter.

### CODE REVIEW

File: examples/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -275,7 +275,7 @@
 			);
 			inputPaths = (
 				"${SRCROOT}/Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh",
-				"${PODS_ROOT}/../../../../bin/cache/artifacts/engine/ios/Flutter.framework",
+				"${PODS_ROOT}/../../../../bin/cache/artifacts/engine/ios-release/Flutter.framework",
```
Comment (tvolkert) [inline examples/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj]: How is this related?

Comment (cbracken) [reply examples/flutter_gallery/ios/Runner.xcodeproj/project.pbxproj]: Oops -- it's not! Reverted.

**Review by tvolkert — APPROVED** [review]
LGTM
